src/main/scala/mesosphere/marathon/api/v2/Validation.scala

```diff
@@ -52,22 +52,28 @@ object Validation {
   }
 
   private def allRuleViolationsWithFullDescription(violation: Violation,
-                                                   parentDesc: Option[String] = None,
-                                                   attachDot: Boolean = true): Set[RuleViolation] = {
+                                                   parentDesc: Option[String] = None): Set[RuleViolation] = {
     def concatPath(parent: String, child: Option[String], attachDot: Boolean): String = {
-      child.map(c => parent + { if (attachDot) "." else "" } + c).getOrElse(parent)
+      child.map(c => parent + c + { if (attachDot) "." else "" }).getOrElse(parent)
     }
 
     violation match {
       case r: RuleViolation => Set(parentDesc.map(p =>
-        r.withDescription(concatPath(p, r.description, attachDot)))
+        r.withDescription(concatPath(p, r.description, attachDot = false)))
         .getOrElse(r))
       case g: GroupViolation => g.children.flatMap { c =>
         parentDesc.map { p =>
-          val desc = concatPath(p, g.description, attachDot)
-          allRuleViolationsWithFullDescription(c, Some(desc), g.description.isDefined)
+          val forceNoDot = g.value match {
+            case x :: xs => false
```

Comment:
Here you pattern match for a non empty list. Perhaps you want to do something like: 
```scala
case _: Iterable => false
```